cudax/include/cuda/experimental/__cufile/detail/batch_handle_impl.hpp

```diff
@@ -0,0 +1,176 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+// This file provides the implementation of batch_handle methods
+// It's included after the class definition to avoid circular dependency issues
+
+#include "cuda/experimental/__cufile/batch_handle.hpp"
+
+namespace cuda::experimental::cufile
+{
+
+// Forward declaration to avoid including file_handle.hpp in batch_handle.hpp
+class file_handle_base;
+
+// batch_io_params_span constructor implementation
+template <typename T>
+batch_io_params_span<T>::batch_io_params_span(
+  cuda::std::span<T> buf, off_t f_off, off_t b_off, CUfileOpcode_t op, void* ck)
+    : buffer(buf)
+    , file_offset(f_off)
+    , buffer_offset(b_off)
+    , opcode(op)
+    , cookie(ck)
+{
+  static_assert(::std::is_trivially_copyable_v<T>, "Type must be trivially copyable for cuFile operations");
+}
+
+// batch_io_result method implementations
+inline bool batch_io_result::is_complete() const noexcept
```

Comment:
Question: We are a header only library, so separating implementation and definition does not really serve anything. Please move in class if its not necessary